demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java

```diff
@@ -0,0 +1,202 @@
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ *                                      derived from HelloWorldServer example
+ ******************************************************************************/
+package org.eclipse.californium.examples;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.californium.core.CoapResource;
+import org.eclipse.californium.core.CoapServer;
+import org.eclipse.californium.core.coap.CoAP;
+import org.eclipse.californium.core.coap.MediaTypeRegistry;
+import org.eclipse.californium.core.coap.Request;
+import org.eclipse.californium.core.coap.Response;
+import org.eclipse.californium.core.network.CoapEndpoint;
+import org.eclipse.californium.core.network.EndpointManager;
+import org.eclipse.californium.core.network.Exchange;
+import org.eclipse.californium.core.network.config.NetworkConfig;
+import org.eclipse.californium.core.server.resources.CoapExchange;
+import org.eclipse.californium.core.server.resources.Resource;
+
+public class SimpleFileServer extends CoapServer {
+	private static final Logger LOG = Logger.getLogger(SimpleFileServer.class.getName());
+
+	private static final int COAP_PORT = NetworkConfig.getStandard().getInt(NetworkConfig.Keys.COAP_PORT);
+	private static final int MAX_FILE_SIZE = NetworkConfig.getStandard().getInt(NetworkConfig.Keys.MAX_RESOURCE_BODY_SIZE);
+
+	/*
+	 * Application entry point.
+	 */
+	public static void main(String[] args) {
+
+		try {
+
+			// create server
+			SimpleFileServer server = new SimpleFileServer();
+			// add endpoints on all IP addresses
+			server.addEndpoints();
+			server.start();
+
+		} catch (SocketException e) {
+			System.err.println("Failed to initialize server: " + e.getMessage());
+		}
+	}
+
+	/**
+	 * Add individual endpoints listening on default CoAP port on all IPv4
+	 * addresses of all network interfaces.
+	 */
+	private void addEndpoints() {
+		for (InetAddress addr : EndpointManager.getEndpointManager().getNetworkInterfaces()) {
+			// only binds to IPv4 addresses and localhost
+			if (addr instanceof Inet4Address || addr.isLoopbackAddress()) {
+				InetSocketAddress bindToAddress = new InetSocketAddress(addr, COAP_PORT);
+				addEndpoint(new CoapEndpoint(bindToAddress));
+			}
+		}
+	}
+
+	/*
+	 * Constructor for a new simple file server. Here, the resources of the
+	 * server are initialized.
+	 */
+	public SimpleFileServer() throws SocketException {
+
+		add(new FileResource("data", new File("data")));
```

Comment:
I know, but the demo apps are intended to be used from the command line (they all end up a single jar file apps in the *run* folder) and therefore IMHO they should support at least the bare minimum config without requiring users to recompile, don't you think?